src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    public static final int DEFAULT_SIZE = 10;
+    public static final double INCREASE = 1.5;
+    private int size;
+    private T[] elementData;
+
+    public ArrayList() {
+        this.elementData = (T[]) new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            increaseArray();
+        }
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
```

Comment:
here we need a different check,   with this condition: if (index < 0 || index > size) {

Reply:
If honestly I do not understand why we need it. 
checkIndex checks index and then works.

if I change "(index < 0 || index > size)" tests don't pass

In checkIndex I have "index < 0 || index >= size" and it works
Why is it wrong?

Reply:
`        if (index < 0 || index > size) {
            throw new ArrayListIndexOutOfBoundsException(YOUR_MESSAGE);
        }
// other logic
`

It should work in such way

Reply:
Oh, thank you, got it
I've thought I need to add it in another block of code))